stackrox/stackrox
# chore(ui): Fix lint errors before eslint-plugin-react upgrade

## Description

### Problem

Because upgrades to some lint packages seem to have a domino effect errors from other packages, fix lint errors in batches as a prerequisite to bulk upgrade.

### Analysis

1. react/forbid-prop-types 2 Prop type "any" is forbidden
    * src/Components/InfoList/InfoList.js
    * src/Components/MultiSelect/MultiSelect.js
2. react/forbid-prop-types 27 Prop type "object" is forbidden
    * 14 in Components folder
    * 2 in Compliance folder
    * 1 in Login folder which we will eventually rewrite in TypeScript
    * 10 in Risk folder
3. react/no-array-index-key 1 Do not use Array index in keys
    * src/Components/PagerControls.js rule now detects `page.toString()`
        `PagerDots` is used only in `Widget` where they were hidden in rox 1364 in 2019-02-14
4. react/no-unknown-property 1 Unknown property 'dataTestId' found
    * src/Containers/Login/TestLoginResultsPage.js `dataTestId="button-close-window"` on HTML `button` element.
5. react/no-unused-prop-types 4 PropType is defined but prop is never used
    * src/Containers/AccessControl/AccessScopes/AccessScopeForm.tsx obsolete props because `useState` hook is in component
        * labelSelectorsEditingState
        * setLabelSelectorsEditingState
    * src/Containers/AccessControl/PermissionSets/PermissionSetsList.tsx obsolete prop because `PermissionSets` handles create
        * handleCreate
    * src/Containers/Collections/CollectionForm.tsx obsolete prop because `CollectionFormDrawer` handles create
        * headerContent
6. react/prop-types 3 missing in props validation
    * src/Components/CheckboxTable.js TODO
        * original
        * original.id
    * src/Containers/ConfigManagement/Entity/widgets/DeploymentsWithFailedPolicies.js
        * original.deployments Solution add property

### Solution

1. react/forbid-prop-types any
    Move each component inline at its one use so `any` becomes moot.
    * src/Components/InfoList/InfoList.js
    * src/Components/MultiSelect/MultiSelect.js deleted props whose values are not in DOM elements nor apparently supported in ReactSelect.tsx file.
2. react/forbid-prop-types object
    * Allow object in eslintrc.js file.
3. react/no-array-index-key
    * Delete `PagerDots` component and hidden `PagerDots` element.
4. react/no-unknown-property
    * Delete unknown `dataTestId` attribute, especially because integration tests do not use it.
5. react/prop-types
    * AccessScopeForm and AccessScopeFormWrapper: comment out unused props and comment about possible bug
    * PermissionSetsList and PermissionSets: delete unused prop
    * CollectionForm: delete unused prop
6. react/prop-types
    * Add eslint disable comment.
    * Add prop type.

### Residue

1. src/Containers/VulnMgmt/List/Cves/CveBulkActionDialogue.js
    * Both `disallowedCves` and second occurrence of `InfoList` in CveBulkActionDialogue.js file might be unreachable after separation of generic `CVE` into `IMAGE_CVE` and so on.
2. src/Containers/VulnMgmt/List/Cves/CveToPolicyShortForm.js
    * Both during accessibility and now lint fixes, I find that policy creation fails because the name is lost.
3. src/Containers/AccessControl/AccessScopes/AccessScopeFormWrapper.tsx
    * Merge with AccessScopeForm.tsx file.

## Checklist

## Testing Performed

1. `yarn lint` in ui
    Before changes:
    * with current version: 0 errors
    * with temporary upgrade to 7.33.2 version: 38 errors in 33 files
    After changes:
    * with temporary upgrade to 7.33.2 version: 0 errors
    * with downgrade to current version: 0 errors
2. `yarn build` in ui and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.js: -33 = 3763713 - 3763746
        total: -2341 = 10064172 - 10066513
    * `ls -al build/static/js/*.js | wc -l`
        0 = 81 - 81 files
3. `yarn start` in ui

### Manual testing

1. Visit /main/vulnerability-management/image-cves: select vulnerabilities, and then click **Add to policy**
    * See vulnerabilities from `cvesToDisplay`
2. Visit /main/compliance and inspect **Passing standards by cluster** widget
    * See pager arrows and hidden pager dots for multiple clusters on stagingdb
    * See pager arrows even for single cluster via temporary code change on local deployment
